inst/views/components/metrics_charts.html

```diff
@@ -188,15 +188,15 @@
     return metric === 'acc' || metric === 'accuracy';
   };
 
-   // helper to format y tick marks. use the default
+  // helper to format y tick marks. use the default
   // d3 formatter but strip long sequences of zeros
   // followed by a single digit at the end (result
   // of JavaScript floating point rouning issues
   // during axis interpolation)
   var default_format = d3.format("");
   this.$y_tick_format = function(d) {
     var fmt = default_format(d);
-    return fmt.replace(/0+\d$/, '');
+    return fmt.replace(/0{4,}[1-9]$/, '0');
```

Comment:
I'm curious why you replace this with a single zero in this case? I don't have the full context but I assume the issue is that d3 is formatting some floats like:

    7.50000000003

Do you want that displayed as 7.5 or 7.50 in this case?

Also, I wonder if you should check if a `.` exists in the formatted number? If none exists you could then just skip the fix-up step. This would also avoid re-writing an axis tick like `100004` (although I guess that probably wouldn't show up in practice).

Reply:
Here's some discussion of the issue: https://stackoverflow.com/questions/1458633/how-to-deal-with-floating-point-number-precision-in-javascript

Replacing with the `0` is so that the following:

```
7.00000003
```

Doesn't get replaced with:

```
7.
```

Good point about the `.`. I'll put in a check to only do this when I see a decimal.